### PR TITLE
Not needed cert and key was not removed on remote

### PR DESCRIPTION
**Pull request by darmach**
Setup RPM remotes task was not removing no longer necessary client cert and key after switch to user/pass auth. Removed default 'omit' that caused this behaviour. When client_cert and client_key were previously configured and role executed with remote_username and remote_password, client_cert and client_key were not removed on on RPM remote, resulting in still attempting to user cert auth.

### CODE REVIEW

File: roles/pulp_repository/tasks/rpm.yml [hunk 1]
```diff
@@ -19,8 +19,8 @@
     validate_certs: "{{ pulp_validate_certs | bool }}"
     name: "{{ pulp_repository_rpm_repos[repository_index].name }}-remote"
     ca_cert: "{{ pulp_repository_rpm_repos[repository_index].ca_cert | default(omit) }}"
-    client_cert: "{{ pulp_repository_rpm_repos[repository_index].client_cert | default(omit) }}"
-    client_key: "{{ pulp_repository_rpm_repos[repository_index].client_key | default(omit) }}"
+    client_cert: "{{ pulp_repository_rpm_repos[repository_index].client_cert | default('') }}"
+    client_key: "{{ pulp_repository_rpm_repos[repository_index].client_key | default('') }}"
```
Comment (markgoddard) [inline roles/pulp_repository/tasks/rpm.yml:23]: ah, I see what happened. Still, I'm not sure this is the right place for the fix.

1. It's good to rely on the module defaults
2. It's nice to be able to provide a subset of parameters and leave the rest as 'don't care' when updating existing remotes
3. This is now inconsistent with other parameters.

I suggest we explicitly set these in stackhpc-kayobe-config.

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: roles/pulp_repository/tasks/rpm.yml (hunk 1 above)
Comment (darmach) [reply roles/pulp_repository/tasks/rpm.yml:23]: Hmm, you mean setting `client_cert` and `client_key` to '' in kayobe-config when switching from cert to password? (To 'force' the role to remove certs from rpm remotes already configured in pulp)

Comment (markgoddard) [reply roles/pulp_repository/tasks/rpm.yml:23]: Yes

Comment (mnasiadka) [reply roles/pulp_repository/tasks/rpm.yml:23]: I think with the ,,wave'' of upgrades, we'll get more people with the same problem - would be good to at least document it or fix it in s-k-c.

Comment (markgoddard) [reply roles/pulp_repository/tasks/rpm.yml:23]: ah, I didn't know that @m-bull. I thought you said it "just worked" :D

Comment (markgoddard) [reply roles/pulp_repository/tasks/rpm.yml:23]: +1 for fix in s-k-c

**Comment by markgoddard** [comment]
Fixed in config instead

**Pull request closed without merging**